src/savegame_json.cpp

```diff
@@ -1566,9 +1566,87 @@ void time_duration::serialize( JsonOut &jsout ) const
     jsout.write( turns_ );
 }
 
+time_duration time_duration::read_from_json_string( JsonIn &jsin )
```

Comment:
This is a fairly complex parsing function, and it should be easy to write tests for it, so I encourage you to do so.